src/types.ts

```diff
@@ -183,10 +197,8 @@ export async function iterativelyCheckStatus(
         );
     } else {
         // Make a recursive call, but with a delay.
-        // Local var `arguments` holds what was passed in the current call
-
-        // eslint-disable-next-line prefer-rest-params
-        setTimeout(iterativelyCheckStatus, CHECK_STATUS_TIMEOUT, ...arguments);
+        await sleep(CHECK_STATUS_TIMEOUT);
+        iterativelyCheckStatus(txHash, starknetWrapper, resolve, reject, retryCount);
```

Comment:
Why use recursion here? Can't all of this be done in a `while true` loop? Iterating in a loop should be computationally cheaper than recursing?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/57b63a3440197fea0fc2600adf637b11376c2d4f